src/pages/VestingClaim.tsx

```diff
@@ -0,0 +1,191 @@
+import { Contract as EthersContract, ethers } from "ethers"
+import React, { ReactElement, useEffect, useState } from "react"
+import { SDL_TOKEN, SDL_TOKEN_ADDRESSES } from "../constants"
+import { Trans, useTranslation } from "react-i18next"
+import { commify, formatBNToString } from "../utils"
+import { notifyCustomError, notifyHandler } from "../utils/notifyHandler"
+import Button from "../components/Button"
+import INVESTOR_EMPLOYEE_VESTING_CONTRACT_ABI from "../constants/abis/vesting.json"
+import SDL_TOKEN_ABI from "../constants/abis/sdl.json"
+import { Sdl } from "../../types/ethers-contracts/Sdl"
+import TopMenu from "../components/TopMenu"
+import { Vesting } from "../../types/ethers-contracts/Vesting"
+import logo from "../assets/icons/logo.svg"
+import plusIcon from "../assets/icons/plus.svg"
+import styles from "../components/TokenClaimModal.module.scss"
+import { useActiveWeb3React } from "../hooks"
+import useAddTokenToMetamask from "../hooks/useAddTokenToMetamask"
+
+function VestingClaim(): ReactElement {
+  const { t } = useTranslation()
+  const { account, chainId, library } = useActiveWeb3React()
+  const { addToken, canAdd } = useAddTokenToMetamask({
+    ...SDL_TOKEN,
+    icon: `${window.location.origin}/logo.svg`,
+  })
+
+  const [isValidBene, setIsValidBene] = useState(false)
+  const [claimableVestedAmount, setClaimableVestedAmount] = useState("0.0")
+  const [remainingAmount, setRemainingAmount] = useState("0.0")
+  const [vestingContract, setVestingContract] = useState<Vesting>()
+
+  useEffect(() => {
+    const fetchBeneficiaries = async () => {
+      if (!library || !chainId) return
+      const provider = new ethers.providers.Web3Provider(
+        window.ethereum as never,
+      )
+      const SDLContract = new EthersContract(
+        SDL_TOKEN_ADDRESSES[chainId],
+        SDL_TOKEN_ABI,
+        provider.getSigner(),
```

Comment:
Prefer to use `account` from `useActiveWeb3React` - check out the `useContract` hook or https://github.com/saddle-finance/saddle-frontend/blob/master/src/hooks/usePoolData.ts#L166 for examples.